### ISSUE TITLE
Handling Null Claims Properly

### ISSUE THREAD

**Issue by Coz1178** [matteofabbri/AspNetCore.Identity.Mongo#56]
With a clean project, I'm getting exceptions after registering a new logon and attempting to sign in.  I've tracked it down to this:

`public async Task<IList<Claim>> GetClaimsAsync(TRole role, CancellationToken cancellationToken = default)
        {
            if (role == null) throw new ArgumentNullException(nameof(role));

            cancellationToken.ThrowIfCancellationRequested();

            var dbRole = await _collection.FirstOrDefaultAsync(x => x.Id == role.Id, cancellationToken: cancellationToken).ConfigureAwait(false);

            return dbRole.Claims.Select(e => new Claim(e.ClaimType, e.ClaimValue)).ToList();
        }`
It seems that when I'm logging on, the Claims object is null, which is throwing the exception.  The dbRole variable populates as expected.  Changing to this seemed to resolve the exception:
`public async Task<IList<Claim>> GetClaimsAsync(TRole role, CancellationToken cancellationToken = default)
        {
            if (role == null) throw new ArgumentNullException(nameof(role));

            cancellationToken.ThrowIfCancellationRequested();

            var dbRole = await _collection.FirstOrDefaultAsync(x => x.Id == role.Id, cancellationToken: cancellationToken).ConfigureAwait(false);

            if (dbRole.Claims == null)
                return new List<Claim>();
            else
                return dbRole.Claims.Select(e => new Claim(e.ClaimType, e.ClaimValue)).ToList();
        }`

**Comment by matteofabbri** [comment]
Current version implement checks on null, which version are you running ?

**Comment by Coz1178** [comment]
The version I pulled from nuget is 6.7.1.  It is providing the below error:

`ArgumentNullException: Value cannot be null. (Parameter 'source')
System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
System.Linq.Enumerable.Select<TSource, TResult>(IEnumerable<TSource> source, Func<TSource, TResult> selector)
AspNetCore.Identity.Mongo.Stores.RoleStore<TRole>.GetClaimsAsync(TRole role, CancellationToken cancellationToken)
Microsoft.AspNetCore.Identity.UserClaimsPrincipalFactory<TUser, TRole>.GenerateClaimsAsync(TUser user)
Microsoft.AspNetCore.Identity.UserClaimsPrincipalFactory<TUser>.CreateAsync(TUser user)
Microsoft.AspNetCore.Identity.SignInManager<TUser>.CreateUserPrincipalAsync(TUser user)
Microsoft.AspNetCore.Identity.SignInManager<TUser>.SignInWithClaimsAsync(TUser user, AuthenticationProperties authenticationProperties, IEnumerable<Claim> additionalClaims)
Microsoft.AspNetCore.Identity.SignInManager<TUser>.SignInOrTwoFactorAsync(TUser user, bool isPersistent, string loginProvider, bool bypassTwoFactor)
Microsoft.AspNetCore.Identity.SignInManager<TUser>.PasswordSignInAsync(TUser user, string password, bool isPersistent, bool lockoutOnFailure)
Microsoft.AspNetCore.Identity.SignInManager<TUser>.PasswordSignInAsync(string userName, string password, bool isPersistent, bool lockoutOnFailure)
Sheetly.Areas.Identity.Pages.Account.LoginModel.OnPostAsync(string returnUrl) in Login.cshtml.cs
+
                var result = await _signInManager.PasswordSignInAsync(Input.sUserName, Input.Password, Input.RememberMe, lockoutOnFailure: true);`

To get around this I have been pulling the source from this git repo and modifying GetClaimsAsync as mentioned above.

**Comment by vova3211** [comment]
@Coz1178 did you do steps from [this migration guide](https://github.com/matteofabbri/AspNetCore.Identity.Mongo/blob/master/docs/MigrationGuideToVersion6_7_0AndUpper.md)?

**Comment by alexanderbikk** [comment]
Hello guys @matteofabbri , @vova3211, @Coz1178 
I've got the same issue when trying to call this method CreateUserPrincipalAsync(user). I did it in custom authorization when generating my token, so the user still not logged in at this stage and has no Claims and I just want to add Claims for him.
I believe this is the same problem that was mentioned above.

Btw I didn't have this issue in 6.3.5 version

Tanks
